org.OpenT2T.Onboarding.VeraHub/node/hub.js

```diff
@@ -0,0 +1,243 @@
+                }
+            }
+        },
+        {
+            type: 'input',
+            name: 'password',
+            message: 'Vera Password (create this in the Vera app): ',
+            validate: function(value) {
+                var pass = !!value;
+                if (pass) {
+                    return true;
+                } else {
+                    return 'Please enter a valid Password.';
+                }
+            }
+        }
+    ],
+
+    hubInfo: undefined,
+    devices: undefined,
+    sessionToken: undefined,
+    relayServer: undefined,
+    pkDevice: undefined,
+
+    connect : function(answers, errorCallback)
+    {
+        internal.pkDevice = answers.pkDevice;
+        internal.username = answers.username;
+        internal.password = answers.password;
+
+        var deferred = q.defer();
+        var myRef = this;
+                var mmsInfoPromise = internal.getMmsAuthInfo();
+                mmsInfoPromise.then( 
+                    function(mmsInfo)
+                    {
+                        var firstSessionPromise = internal.getSessionToken(internal.mmsInfo.Server_Account);
+                        firstSessionPromise.then(
+                            function(sessionToken)
+                            {
+                                var deviceInfoPromise = internal.getDeviceInfo(mmsInfo.Server_Account, sessionToken, answers.pkDevice);
+                                deviceInfoPromise.then(
+                                    function (deviceInfo)
+                                    {
+                                        var relaySessionPromise = internal.getSessionToken(deviceInfo.Server_Relay, mmsInfo);
+                                        relaySessionPromise.then(
+                                            function (relaySessionToken)
+                                            {
+                                                var devicesPromise = internal.getUserData(deviceInfo.Server_Relay, relaySessionToken, answers.pkDevice);
+                                                devicesPromise.then(
+                                                    function(hubInfo)
+                                                    {
+                                                        myRef.sessionToken = relaySessionToken;
+                                                        myRef.relayServer = deviceInfo.Server_Relay;
+                                                        myRef.pkDevice = answers.pkDevice;
+                                                        myRef.hubInfo = hubInfo;
+                                                        myRef.devices = hubInfo.devices;
+                                                        deferred.resolve();
+                                                    },
+                                                    function (error){ handleError(errorCallback, error);}        
```

Comment:
ah ok. It is in index.js. But this is sort of circular dependency. index.js requires hub.js. but not the other way. So, you may want to move handleError here ?

---
In reply to: [69948404](https://github.com/openT2T/onboarding/pull/5#discussion_r69948404) [](ancestors = 69948404)